### PR TITLE
reconfigure eslint and fix errors

### PR DESCRIPTION
**Pull request by dmaish**
#### What does this PR do?
Introduces eslint rules that make the codebase uniform and readable

#### Description of Task to be completed?
enable the following eslint rules:

- end-of-line
- no-unused variables
- lines not exceeding maximum of 100
- no trailling spaces
- spaces after opening and closing braces

#### How should this be manually tested?
- After pulling latest changes from develop, run ```yarn install```
- When a trailling or unused variable is introduced, esling should prompt an error about that

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[161323203](https://www.pivotaltracker.com/story/show/161323203)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "env": {
+        "node": true,
+        "es6": true,
+        "jest": true
+    },
+    "parserOptions": {
+        "ecmaVersion": 6,
+        "sourceType": "module"
+    },
+    "rules": {
```
Comment (ezrqnkemboi) [inline .eslintrc.json]: I like the new rules set up here.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "serve": "node dist/index.js",
     "start": "NODE_ENV=production yarn run build && yarn run serve",
     "start:dev": "NODE_ENV=development  DEBUG=log  nodemon --exec babel-node src/index.js",
-    "lint": "eslint src/**/**.js",
+    "lint": "node_modules/.bin/eslint ./src",
```
Comment (ezrqnkemboi) [inline package.json]: I think this implementation is good as during yarn install, the eslint will be installed, which will be used for linting process.

**Comment by samuelkubai** [comment]
@dmaish please follow the correct commit conventions as defined on the playbook.

**Comment by samuelkubai** [comment]
@dmaish please fix the PR title and the conflicts you have.

**Comment by dmaish** [comment]
> @dmaish please follow the correct commit conventions as defined on the playbook.

issues fixed

**Comment by dmaish** [comment]
> > @dmaish please follow the correct commit conventions as defined on the playbook.
> 
> issues fixed

@samuelkubai

**Pull request merged**